src/seeds.gratitude.cpp

```diff
@@ -70,38 +92,88 @@ ACTION gratitude::acknowledge (name from, name to, string memo) {
         item.receivers.push_back(to);
     });
   }
+
+  // Updates ack stats
+  auto stitr = stats.rbegin();
+  auto round_id = stitr->round_id;
+
+  auto stitr2 = stats.find(round_id);
+  auto oldacks = stitr2->num_acks;
+  stats.modify(stitr2, _self, [&](auto& item) {
+      item.num_acks = oldacks + 1;
+  });
+
 }
 
+
 ACTION gratitude::testacks() {
   require_auth(get_self());
 
   auto actr = acks.begin();
+
   while (actr != acks.end()) {
-    calc_acks(actr->donor);
-    actr = acks.erase(actr);
+    _calc_acks(actr->donor);
+    actr++;
   }
 }
 
-ACTION gratitude::newround() {
+ACTION gratitude::calcacks(uint64_t start) {
+  require_auth(get_self());
+
+  auto actr = start == 0 ? acks.begin() : acks.find(start);
+
+  _calc_acks(actr->donor);
+
+  actr++;
+
```

Comment:
Normally, we process a bunch of items at the same time, e.g. 40 or whatever, 1 at a time is slow?!

Reply:
In this case, each account can have around 10 transfers, so I was thinking of doing one account at a time, but we may batch 10 to test.

Reply:
oh thats ok then